### PR TITLE
Show refdes in block diagram viewer

### PR DESCRIPTION
**Pull request by ducky64**
Show the refdes (for applicable blocks) in the block viewer, in addition to the currently shown block name. If the block just contains a single inner block with a refdes, shows that - this helps deal with the heavy use of the adapter pattern (eg, DecouplingCapacitor contains a Capacitor, that Capacitor is what has the refdes but it makes sense to also show it on the wrapping DecouplingCapacitor).

This is done by using the mapper infrastructure provided in the EdgirGraph -> ElkGraph pass, as an additional mapper that adds a Title field to each ElkNode, which the conversion pass then reads and uses for the title (if present). Mappers run for each block, checking for a refdes (in the block, or an innermost only-child block), and if it exists creates a title with it. Architecture-wise this is probably the cleanest option, somewhat analogous to how DesignToolTipMap is structured.

This also adds the mapper label infrastructure to ports, though that is not currently used.

Works with both the current block visualizer panel and the PDFing pass. I'd expect this to work with the recursive PDFing when that gets merged, though this will probably add a few more lines to that merge conflict 😅 

Other minor cleanup: remove the unneeded Project parameter from DesignToolTipTextMap, use DesignPath.lastString instead of reimplementing it.

### CODE REVIEW

File: src/main/scala/edg_ide/edgir_graph/ElkEdgirGraphUtils.scala [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+import edgir.elem.elem
 import org.eclipse.elk.graph.{ElkGraphElement, ElkNode}
 
 import scala.jdk.CollectionConverters._
 
 
 object ElkEdgirGraphUtils {
   import org.eclipse.elk.graph.properties.IProperty
+
+  // Adds the DesignPatnProperty, containing the node's DesignPath, to nodes.
```
Comment (TayKaiJun) [inline src/main/scala/edg_ide/edgir_graph/ElkEdgirGraphUtils.scala:17]: Is this supposed to spell DesignPathProperty?

File: src/main/scala/edg_ide/runner/CompileProcessHandler.scala [hunk 2]
```diff
@@ -297,10 +297,7 @@ class CompileProcessHandler(project: Project, options: DesignTopRunConfiguration
           indicator.setText("EDG compiling: generating PDF")
           console.print("Printing PDF\n", ConsoleViewContentType.LOG_INFO_OUTPUT)
           val (pdfGeneration, pdfTime) = timeExec {
-            PDFGeneratorUtil.generate(
-            HierarchyGraphElk.HBlockToElkNode(compiled.getContents,
-              mappers=Seq(new ElkEdgirGraphUtils.TitleMapper(compiler))),
-            options.pdfFile)
+            PDFGeneratorUtil.generate(compiled.getContents, mappers=Seq(new ElkEdgirGraphUtils.TitleMapper(compiler)),options.pdfFile)
```
Comment (TayKaiJun) [inline src/main/scala/edg_ide/runner/CompileProcessHandler.scala:300]: @ducky64 is this the right way for me to use the new refdes name mapping feature?

File: src/main/scala/edg_ide/edgir_graph/ElkEdgirGraphUtils.scala (hunk 1 above)
Comment (ducky64) [reply src/main/scala/edg_ide/edgir_graph/ElkEdgirGraphUtils.scala]: It definitely was 🙃
Good catch!

File: src/main/scala/edg_ide/runner/CompileProcessHandler.scala (hunk 2 above)
Comment (ducky64) [reply src/main/scala/edg_ide/runner/CompileProcessHandler.scala:300]: Yep! Looks like it works too!

**Review by ducky64 — COMMENTED** [review]
fixes from review!

**Review by TayKaiJun — APPROVED** [review]
Lgtm

**Pull request merged**